src/lib.rs

```diff
@@ -76,32 +196,78 @@ impl TempDir {
                        "too many temporary directories already exist"))
     }
 
-    /// Attempts to make a temporary directory inside of `env::temp_dir()` whose
-    /// name will have the prefix `prefix`. The directory will be automatically
-    /// deleted once the returned wrapper is destroyed.
+    /// Access the [`Path`] to the temporary directory.
     ///
-    /// If no directory can be created, `Err` is returned.
-    pub fn new(prefix: &str) -> io::Result<TempDir> {
-        TempDir::new_in(&env::temp_dir(), prefix)
+    /// [`Path`]: http://doc.rust-lang.org/std/path/struct.Path.html
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use tempdir::TempDir;
+    ///
+    /// let tmp_dir = TempDir::new("example").expect("create temp dir");
+    ///
+    /// // Check that the temp directory actually exists.
+    /// assert!(tmp_dir.path().exists());
+    /// ```
+    pub fn path(&self) -> &path::Path {
+        self.path.as_ref().unwrap()
     }
 
-    /// Unwrap the wrapped `std::path::Path` from the `TempDir` wrapper.
-    /// This discards the wrapper so that the automatic deletion of the
-    /// temporary directory is prevented.
+    /// Unwrap the [`Path`] contained in the `TempDir` and
```

Comment:
Here you write "unwrap", but just above you wrote "Attempts". Any reason that it doesn't use third singular pronum here?